src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,125 @@
+
+
+
+/**
+ * Saves and stores tasks that user has input.
+ */
+public class Storage {
+    private String database;
+    private List<Task> tasks = new ArrayList<>();
+
+    /**
+     * Reads the text file 'database.txt' which contains the saved tasks user input previously.
+     *
+     * @return List of tasks which user had from the last time they used the program.
+     * @throws DukeException throw when error occurs.
+     */
+    public List<Task> readFile() throws DukeException {
+        String currentDirectory = System.getProperty("user.dir");
+        File dataFolder = new File(currentDirectory + File.separator + "data");
+        boolean directoryExists = dataFolder.exists() && dataFolder.isDirectory();
+        if (directoryExists) {
+            File database = new File(dataFolder.getAbsolutePath() + File.separator + "database.txt");
+            try {
+                Scanner s = new Scanner(database);
+                while (s.hasNext()) {
+                    String task = s.nextLine();
+                    convertIntoTasks(task);
+                }
+                this.database = database.getAbsolutePath();
+                return tasks;
+            } catch (FileNotFoundException e) {
+                throw new DukeException("No file found sorry please make one");
+            }
+        } else {
+            dataFolder.mkdir();
+            File database = new File(dataFolder.getAbsolutePath() + File.separator + "database.txt");
+            try {
+                database.createNewFile();
+                this.database = database.getAbsolutePath();
+                return tasks;
+            } catch (IOException e) {
+                throw new DukeException("File already exists");
+            }
+        }
+    }
+
+    private void convertIntoTasks(String s) {
+        Task t;
+        String[] descriptions = s.split("\\|");
+        if (descriptions[0].equals("T")) {
+            t = new Todo(descriptions[2]);
+            if (descriptions[1].equals("T")) {
+                t.markAsDone();
+            }
+            tasks.add(t);
+        } else if (descriptions[0].equals("D")) {
+            t = new Deadline(descriptions[2], descriptions[3]);
+            if (descriptions[1].equals("T")) {
+                t.markAsDone();
+            }
+            tasks.add(t);
+        } else if (descriptions[0].equals("E")) {
+            t = new Event(descriptions[2], descriptions[3]);
+            if (descriptions[1].equals("T")) {
+                t.markAsDone();
+            }
+            tasks.add(t);
+        }
+    }
+
+    /**
+     * Writes the list of tasks into text file 'database.txt' and saves it for future use.
+     *
+     * @param tasks current list of tasks of user.
+     * @throws IOException throw when FileWriter operations fail.
+     * @throws DukeException throw when error occurs and need to print error message.
+     */
+    public void save(List<Task> tasks) throws IOException, DukeException {
```

Comment:
Minor nit with the method name here, but perhaps `saveToFile` would be more specific as to what this method actually does?